### DISCUSSION TITLE
automatically find meson_version for existing project

### DISCUSSION THREAD

**Discussion by LunarLambda** [mesonbuild/meson#11254] [Ideas]
Meson, like CMake, allows specifying a version requirement to gracefully error if a user is on an unsupported version. While not explicitly stated so, this is very much a "best practice" feature

However, unless one is *very* meticulous in reading the documentation when writing build definitions, it's probably pretty hard to find what the minimum required version actually *is*. My projects hover somewhere between 0.56 and 0.60 because of wrap system features, but that took me a good bit to figure out. (The documentation does list most if not all version-relevant information, but it can be hard to pick out of the rest of the text sometimes)

so, would it be possible to add a command that analyses your meson.build file and figures out the version requirement?

It could be integrated into the meson file rewriter too, and what would be even sweeter, if it also tells the user what features determined the minimum version, in case you find its too high and the feature is something that could be lived without

**Comment by eli-schwartz** [comment]
What I usually do there is to add `meson_version: '>=0.1.0'`, try to build the project, and see what warnings get triggered. Then update that to the highest version on the list of generated warnings.

By setting the version unrealistically low, you're guaranteed to trigger every possible FeatureNew check, and then you can decide which ones to refactor and which ones to document in meson_version.

**Reply by LunarLambda** [reply]
That works. thank you!
